### ISSUE TITLE
aws_elastic_beanstalk_configuration_template.options changes not updated

### ISSUE THREAD

**Issue by atward** [hashicorp/terraform#6301]
Updates/changes to aws_elastic_beanstalk_configuration_template.options are not reflected in AWS. Testing has shows that changes to options delete them from the configuration template. 
There appears to be a clear typo bug on https://github.com/hashicorp/terraform/blob/master/builtin/providers/aws/resource_aws_elastic_beanstalk_configuration_template.go#L174 - but patching that line does not fix the issue.

This is probably related to related: #6035 @catsby

**Comment by atward** [comment]
Relevant lines from debug logs.
Running against patch suggested above: https://github.com/atward/terraform/commit/9e335bc4f8de1ab43d7570d91265d5e74bc02a97
Setting change from
```
    setting {
        namespace = "aws:elb:loadbalancer"
        name = "CrossZone"
        value = "true"
    }
```
to
```
    setting {
        namespace = "aws:elb:loadbalancer"
        name = "CrossZone"
        value = "false"
    }
```
```
2016/04/22 17:13:56 [DEBUG] terraform-provider-aws: 2016/04/22 17:13:56 [DEBUG] [aws-sdk-go] DEBUG: Request elasticbeanstalk/UpdateConfigurationTemplate Details:
2016/04/22 17:13:56 [DEBUG] terraform-provider-aws:
2016/04/22 17:13:56 [DEBUG] terraform-provider-aws: ---[ REQUEST POST-SIGN ]-----------------------------
2016/04/22 17:13:56 [DEBUG] terraform-provider-aws: POST / HTTP/1.1
2016/04/22 17:13:56 [DEBUG] terraform-provider-aws: Host: elasticbeanstalk.ap-southeast-2.amazonaws.com
2016/04/22 17:13:56 [DEBUG] terraform-provider-aws:
2016/04/22 17:13:56 [DEBUG] terraform-provider-aws: Action=UpdateConfigurationTemplate&ApplicationName=myapp&OptionSettings.member.1.Namespace=aws%3Aelb%3Aloadbalancer&OptionSettings.member.1.OptionName=CrossZone&OptionSettings.member.1.Value=false&OptionsToRemove.member.1.Namespace=aws%3Aelb%3Aloadbalancer&OptionsToRemove.member.1.OptionName=CrossZone&TemplateName=apptmpl&Version=2010-12-01


2016/04/22 17:13:57 [DEBUG] terraform-provider-aws: 2016/04/22 17:13:57 [DEBUG] [aws-sdk-go] DEBUG: Response elasticbeanstalk/UpdateConfigurationTemplate Details:
2016/04/22 17:13:57 [DEBUG] terraform-provider-aws: ---[ RESPONSE ]--------------------------------------
2016/04/22 17:13:57 [DEBUG] terraform-provider-aws: HTTP/1.1 200 OK
```

**Comment by atward** [comment]
Appears the code needs to exclude the intersection of os/ns OptionSettings from the remove set - these should an update set, and included in the add set.

**Comment by catsby** [comment]
Thanks! Fixed in #6307
